devtools/gradle/src/main/java/io/quarkus/gradle/builder/QuarkusModelBuilder.java

```diff
@@ -0,0 +1,285 @@
+                    .filter(dep -> dep.getGroupId().equals(d.getModuleGroup()) && dep.getName().equals(d.getModuleName()))
+                    .map(this::getDeploymentArtifact)
+                    .filter(Objects::nonNull)
+                    .forEach(extensions::add);
+            final Set<ResolvedDependency> resolvedChildren = d.getChildren();
+            if (!resolvedChildren.isEmpty()) {
+                extensions.addAll(getDirectExtensionDependencies(resolvedChildren, appDependencies));
+            }
+        }
+        return extensions;
+    }
+
+    private org.gradle.api.artifacts.Dependency getDeploymentArtifact(Dependency dependency) {
+        for (File file : dependency.getPaths()) {
+            if (!file.exists()) {
+                continue;
+            }
+            Properties depsProperties;
+            if (file.isDirectory()) {
+                Path quarkusDescr = file.toPath()
+                        .resolve(BootstrapConstants.META_INF)
+                        .resolve(BootstrapConstants.DESCRIPTOR_FILE_NAME);
+                if (!Files.exists(quarkusDescr)) {
+                    continue;
+                }
+                depsProperties = QuarkusModelHelper.resolveDescriptor(quarkusDescr);
+            } else {
+                try (FileSystem artifactFs = FileSystems.newFileSystem(file.toPath(), getClass().getClassLoader())) {
+                    Path quarkusDescr = artifactFs.getPath(BootstrapConstants.META_INF)
+                            .resolve(BootstrapConstants.DESCRIPTOR_FILE_NAME);
+                    if (!Files.exists(quarkusDescr)) {
+                        continue;
+                    }
+                    depsProperties = QuarkusModelHelper.resolveDescriptor(quarkusDescr);
+                } catch (IOException e) {
+                    throw new GradleException("Failed to process " + file, e);
+                }
+            }
+            String value = depsProperties.getProperty(BootstrapConstants.PROP_DEPLOYMENT_ARTIFACT);
+            String[] split = value.split(":");
+            return new DefaultExternalModuleDependency(split[0], split[1], split[2], null);
+        }
+        return null;
+    }
+
+    private Set<Dependency> collectExtensionDependencies(Project project,
+            Collection<org.gradle.api.artifacts.Dependency> extensions) {
+        final Set<Dependency> platformDependencies = new HashSet<>();
+
+        final Configuration deploymentConfig = project.getConfigurations()
+                .detachedConfiguration(extensions.toArray(new org.gradle.api.artifacts.Dependency[0]));
+        final ResolvedConfiguration rc = deploymentConfig.getResolvedConfiguration();
+        for (ResolvedArtifact a : rc.getResolvedArtifacts()) {
+            if (!isDependency(a)) {
+                continue;
+            }
+
+            final Dependency dependency = toDependency(a, deploymentConfig.getName());
+            platformDependencies.add(dependency);
```

Comment:
It looks like you are not checking for potential conflicts with the user dependencies. That's what I used the versionMap for.

Reply:
I though it was solved using the `alignVersion`

Reply:
Yes, I found it later in another QuarkusModelHelper.